projects/plugins/jetpack/functions.photon.php

```diff
@@ -3,339 +3,79 @@
-		$photon_url .= '?q=' . rawurlencode( $image_url_parts['query'] );
-	}
-
-	if ( $args ) {
-		if ( is_array( $args ) ) {
-			$photon_url = add_query_arg( $args, $photon_url );
-		} else {
-			// You can pass a query string for complicated requests but where you still want CDN subdomain help, etc.
-			$photon_url .= '?' . $args;
-		}
-	}
-
-	if ( isset( $image_url_parts['scheme'] ) && 'https' === $image_url_parts['scheme'] ) {
-		$photon_url = add_query_arg( array( 'ssl' => 1 ), $photon_url );
-	}
-
-	return jetpack_photon_url_scheme( $photon_url, $scheme );
+	_deprecated_function( __FUNCTION__, 'jetpack-12.1', 'Automattic\Jetpack\Image_CDN\Image_CDN_Core::cdn_url' );
+	return Image_CDN_Core::cdn_url( $image_url, $args, $scheme );
 }
 
-/**
- * Add an easy way to photon-ize a URL that is safe to call even if Jetpack isn't active.
- *
- * See: https://jetpack.com/2013/07/11/photon-and-themes/
- */
-add_filter( 'jetpack_photon_url', 'jetpack_photon_url', 10, 3 );
-
-/**
- * WordPress.com
- *
- * If a cropped WP.com-hosted image is the source image, have Photon replicate the crop.
- */
-add_filter( 'jetpack_photon_pre_args', 'jetpack_photon_parse_wpcom_query_args', 10, 2 );
-
 /**
  * Parses WP.com-hosted image args to replicate the crop.
  *
+ * @deprecated 12.1 Use Automattic\Jetpack\Image_CDN\Image_CDN_Core::parse_wpcom_query_args instead.
  * @param mixed  $args Args set during Photon's processing.
  * @param string $image_url URL of the image.
  * @return array|string Args for Photon to use for the URL.
  */
 function jetpack_photon_parse_wpcom_query_args( $args, $image_url ) {
-	$parsed_url = wp_parse_url( $image_url );
-
-	if ( ! $parsed_url ) {
-		return $args;
-	}
-
-	$image_url_parts = wp_parse_args(
-		$parsed_url,
-		array(
-			'host'  => '',
-			'query' => '',
-		)
-	);
-
-	if ( '.files.wordpress.com' !== substr( $image_url_parts['host'], -20 ) ) {
-		return $args;
-	}
-
-	if ( empty( $image_url_parts['query'] ) ) {
-		return $args;
-	}
-
-	$wpcom_args = wp_parse_args( $image_url_parts['query'] );
-
-	if ( empty( $wpcom_args['w'] ) || empty( $wpcom_args['h'] ) ) {
-		return $args;
-	}
-
-	// Keep the crop by using "resize".
-	if ( ! empty( $wpcom_args['crop'] ) ) {
-		if ( is_array( $args ) ) {
-			$args = array_merge( array( 'resize' => array( $wpcom_args['w'], $wpcom_args['h'] ) ), $args );
-		} else {
-			$args = 'resize=' . rawurlencode( absint( $wpcom_args['w'] ) . ',' . absint( $wpcom_args['h'] ) ) . '&' . $args;
-		}
-	} elseif ( is_array( $args ) ) {
-		$args = array_merge( array( 'fit' => array( $wpcom_args['w'], $wpcom_args['h'] ) ), $args );
-	} else {
-		$args = 'fit=' . rawurlencode( absint( $wpcom_args['w'] ) . ',' . absint( $wpcom_args['h'] ) ) . '&' . $args;
-	}
-
-	return $args;
+	_deprecated_function( __FUNCTION__, 'jetpack-12.1', 'Automattic\Jetpack\Image_CDN\Image_CDN_Core::parse_wpcom_query_args' );
```

Comment:
Those will need to be updated to 12.2. You can use `$$next-version$$` here to not have to worry about versions in your PRs. During release, `$$next-version$$` will be automatically replaced by the right number.

```suggestion
	_deprecated_function( __FUNCTION__, 'jetpack-$$next-version$$', 'Automattic\Jetpack\Image_CDN\Image_CDN_Core::parse_wpcom_query_args' );
```